srcs/app/server/Connection.cpp

```diff
@@ -3,24 +3,24 @@
 #include <HttpRequestHandler.hpp>
 
 Connection::Connection(const ServerConfig &server_config, int socket)
-	: server_config_(server_config), socket_(socket),
-		ready_for_response_(false), keep_alive_(true) {
+	: server_config_(server_config), socket_(socket), keep_alive_(true) {
 }
 
-bool	Connection::ReadRequest() {
+enum ReadRequestStatus::Type	Connection::ReadRequest() {
 	char	buffer[4096];
 
 	int nbytes = recv(socket_, buffer, sizeof(buffer), 0);
-	if (nbytes <= 0)
-		return false;
-	ready_for_response_ = true;
+	if (nbytes <= 0) {
+		return ReadRequestStatus::kFail;
+	}
+	ReadRequestStatus::Type status = raw_request_.empty() ?
+									ReadRequestStatus::kStart :
+									ReadRequestStatus::kSuccess;
 	raw_request_.append(&buffer[0], &buffer[nbytes]);
-	return true;
+	return status;
 }
 
-bool	Connection::SendResponse() {
-	if (!ready_for_response_)
-		return true;
+enum SendResponseStatus::Type	Connection::SendResponse() {
```

Comment:
(See below comment before to understand this)

```suggestion
SendResponseStatus::Type	Connection::SendResponse() {
```